### PR TITLE
Configurable tmpDir and context hash in compiled templates

### PR DESCRIPTION
**Pull request by MartinMystikJonas**

### CODE REVIEW

File: src/Template/Component.php [hunk 1]
```diff
@@ -28,4 +29,9 @@ public function getTypeAsString(): string
     {
         return $this->type->describe(VerbosityLevel::typeOnly());
     }
+
+    public function jsonSerialize(): mixed
+    {
+        return [$this->name, $this->getTypeAsString()];
```
Comment (lulco) [inline src/Template/Component.php]: I would prefer to use also some keys in arrays 'name' => $this->name, 'type' => $this->getTypeAsString()

File: src/Template/Template.php [hunk 2]
```diff
@@ -45,4 +47,9 @@ public function getComponents(): array
     {
         return $this->components;
     }
+
+    public function jsonSerialize(): mixed
+    {
+        return [$this->path, $this->variables, $this->components];
```
Comment (lulco) [inline src/Template/Template.php]: keys

File: src/Compiler/LatteToPhpCompiler.php [hunk 3]
```diff
@@ -56,6 +60,37 @@ public function compile(Scope $scope, string $templateContent, array $variables,
...
     }
 
+    /**
+     * @param Variable[] $variables
+     * @param Component[] $components
+     */
+    public function compileFile(Scope $scope, string $templatePath, array $variables, array $components): string
+    {
+        $templateContent = file_get_contents($templatePath) ?: '';
+        $phpContent = $this->compile($scope, $templateContent, $variables, $components);
```
Comment (lulco) [inline src/Compiler/LatteToPhpCompiler.php:70]: compile should be private then

File: src/Rule/LatteTemplatesRule.php [hunk 4]
```diff
@@ -43,7 +43,7 @@ public function __construct(
         FileAnalyserFactory $fileAnalyserFactory,
         RuleRegistry $rulesRegistry,
         CollectorsRegistry $collectorsRegistry,
-        ErrorBuilder $errorBuilder
+        ErrorBuilder $errorBuilder,
```
Comment (lulco) [inline src/Rule/LatteTemplatesRule.php]: we are on php 7.4

File: src/Template/Variable.php [hunk 5]
```diff
@@ -33,4 +34,9 @@ public function getTypeAsString(): string
     {
         return $this->type->describe(VerbosityLevel::typeOnly());
     }
+
+    public function jsonSerialize(): mixed
+    {
+        return [$this->name, $this->getTypeAsString()];
```
Comment (lulco) [inline src/Template/Variable.php]: keys

**Comment by lulco** [comment]
also check the pipeline

File: src/Compiler/LatteToPhpCompiler.php (hunk 3 above)
Comment (MartinMystikJonas) [reply src/Compiler/LatteToPhpCompiler.php:70]: It is used in tests, I would keep it public

**Comment by MartinMystikJonas** [comment]
@lulco Pipeline fixed

File: src/Compiler/LatteToPhpCompiler.php (hunk 3 above)
Comment (lulco) [reply src/Compiler/LatteToPhpCompiler.php:70]: Then we should create some other service just to store files, compiler just creates php content and that new service will just save it to file.

Comment (MartinMystikJonas) [reply src/Compiler/LatteToPhpCompiler.php:70]: It seems as unnecessary complication for me.

**Pull request merged**